operator/pkg/ciliumendpointslice/reconciler.go

```diff
@@ -7,93 +7,209 @@ import (
 	"context"
 
 	"github.com/sirupsen/logrus"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
-	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/operator/metrics"
+	operatorOption "github.com/cilium/cilium/operator/option"
+	"github.com/cilium/cilium/pkg/k8s"
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	cilium_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
 	clientset "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/resource"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 )
 
 // reconciler is used to sync the current (i.e. desired) state of the CESs in datastore into current state CESs in the k8s-apiserver.
 // The source of truth is in local datastore.
 type reconciler struct {
+	logger     logrus.FieldLogger
 	client     clientset.CiliumV2alpha1Interface
 	cesManager operations
+	cepStore   resource.Store[*cilium_v2.CiliumEndpoint]
+	cesStore   resource.Store[*cilium_v2a1.CiliumEndpointSlice]
 }
 
 // newReconciler creates and initializes a new reconciler.
-func newReconciler(client clientset.CiliumV2alpha1Interface, cesMgr operations) *reconciler {
+func newReconciler(client clientset.CiliumV2alpha1Interface, cesMgr operations,
+	logger logrus.FieldLogger, ciliumEndpoint resource.Resource[*cilium_v2.CiliumEndpoint], ciliumEndpointSlice resource.Resource[*cilium_v2a1.CiliumEndpointSlice]) *reconciler {
+	cepStore, _ := ciliumEndpoint.Store(context.Background())
```

Comment:
This is called on Start - CES Controller creates reconciler (using newReconciler).
I moved this to Start just because of this.

I see that newReconciler may be confusing. I don't have much better idea as it doesn't do anything else except for getting those stores, logger and client.

Reply:
Ah sorry, I missed that it was called from a start hook. In that case I don't have any problem with this. I think the naming etc. is fine here. I should've checked how it was used and not assume.